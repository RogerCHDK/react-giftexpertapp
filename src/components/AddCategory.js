import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        // console.log(e.target.value); //asi obtenemos el valor del input en cuanto cambia
        setInputValue(e.target.value); //despues le asignamos ese valor para que lo renderize
    };

    const handleSubmit = (e) => {
        e.preventDefault(); //evito que se refresque la pagina
        if (inputValue.trim().length > 2) { //valido que no este vacio el input
            setCategories( categories => [ inputValue, ...categories]); //obtengo el valor de estado actual, y despues agrego lo que esta en el iput
            setInputValue(''); //Limpio el input
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input 
                type= "text"
                value= {inputValue}
                onChange={ handleInputChange }/>
        </form>
    );
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}