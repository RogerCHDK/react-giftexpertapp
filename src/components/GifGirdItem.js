import React from 'react'
import PropTypes from 'prop-types';

export const GifGirdItem = ( {title, url} ) => {
    return (
        <div className="card animate__animated animate__bounce">
            <img src={ url } alt= {title} />
            <p>{ title }</p> 
        </div>
    )
}

GifGirdItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  };
