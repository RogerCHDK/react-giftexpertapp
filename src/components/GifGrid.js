// Version usando custom hooks
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGirdItem } from './GifGirdItem';


export const GifGrid = ({ category }) => {

    const { loading, data:images } = useFetchGifs(category);
    return (
        <> 
         <h3 className="animate__animated animate__bounce"> { category } </h3>
         { loading && <p className="animate__animated animate__flash">Loading</p>}
         <div className="card-grid">
                { images.map( img => 
                <GifGirdItem 
                key={ img.id }
                { ...img }/> //hack, puedo mandar las propiedades individuales usando el operador spread, sino tendria que hacer: props.img.id
                ) }
        </div>
        </>
    )
}







/*
Primera version funcionanado, usando useEffect
import React, { useEffect, useState } from 'react'
import { getGifs } from '../helpers/getGifs';
import { GifGirdItem } from './GifGirdItem';

export const GifGrid = ({ category }) => {

    useEffect( () => {
        getGifs( category ).then(
            setImages
        )
    }, [ category ]); //esto hace que solo se ejecute una sola vez, aunque se vuelva a renderizar el hook, solo cambia en caso de que la categpria llegue a cambiar

    const [images, setImages] = useState([]);
    
    return (
        <> 
         <h3> { category } </h3>
        <div className="card-grid">
                { images.map( img => 
                <GifGirdItem 
                key={ img.id }
                { ...img }/> //hack, puedo mandar las propiedades individuales usando el operador spread, sino tendria que hacer: props.img.id
                ) }
        </div>
        </>
    )
}*/
