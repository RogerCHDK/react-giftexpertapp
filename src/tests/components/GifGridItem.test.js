import { shallow } from 'enzyme';
import React from 'react';
import { GifGirdItem } from "../../components/GifGirdItem";

describe('Pruebas en <GifGridItem />', () => {
    
    test('should to show the component correctly', () => {
        
        const wrapper = shallow(<GifGirdItem/>);

        expect( wrapper ).toMatchSnapshot();
    });
});