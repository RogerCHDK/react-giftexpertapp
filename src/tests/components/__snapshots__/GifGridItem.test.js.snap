// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> should to show the component correctly 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img />
  <p />
</div>
`;
